src/runtime/threads/policies/parse_affinity_options.cpp

```diff
@@ -842,6 +843,72 @@ namespace hpx { namespace threads { namespace detail
     }
 
     ///////////////////////////////////////////////////////////////////////////
+    void decode_numabalanced_distribution(hwloc_topology_info& t,
+        std::vector<mask_type>& affinities,
+        std::size_t used_cores, std::size_t max_cores,
+        std::vector<std::size_t>& num_pus, error_code& ec)
+    {
+        std::size_t num_threads = affinities.size();
+        num_pus.resize(num_threads);
+        // numa nodes
+        std::size_t num_numas = (std::max)(std::size_t(1),t.get_number_of_numa_nodes());
+        std::vector<std::size_t> num_cores_numa(num_numas, 0);
+        std::vector<std::size_t> num_threads_numa(num_numas, 0);
+        std::size_t cores_t = 0;
+        for (std::size_t n=0; n<num_numas; ++n) {
+            num_cores_numa[n] = t.get_number_of_numa_node_cores(n);
+            cores_t += num_cores_numa[n];
+        }
+        // how many threads should go on each domain
+        for (std::size_t n=0; n<num_numas; ++n) {
+            num_threads_numa[n] = num_threads*num_cores_numa[n]/cores_t;
```

Comment:
This needs to be rounded up, in the case where, for example, 1 thread is requested, we end up with 0 threads per numa domain.